source/extensions/filters/http/ext_proc/processor_state.h

```diff
@@ -187,19 +187,27 @@ class ProcessorState : public Logger::Loggable<Logger::Id::ext_proc> {
   const QueuedChunk& consolidateStreamedChunks() { return chunk_queue_.consolidate(); }
   bool queueOverHighLimit() const { return chunk_queue_.bytesEnqueued() > bufferLimit(); }
   bool queueBelowLowLimit() const { return chunk_queue_.bytesEnqueued() < bufferLimit() / 2; }
-  bool shouldRemoveContentLength() const {
-    // Always remove the content length in 3 cases below:
+  bool shouldRemoveContentLength(const bool retain_content_length_header) const {
+    // Always remove the content length in 4 cases below:
     // 1) STREAMED BodySendMode
     // 2) BUFFERED_PARTIAL BodySendMode
     // 3) BUFFERED BodySendMode + SKIP HeaderSendMode
     // 4) FULL_DUPLEX_STREAMED BodySendMode
     // In these modes, ext_proc filter can not guarantee to set the content length correctly if
     // body is mutated by external processor later.
+    // An exception to the rules outlined above is whenever an external processor server is
+    // consuming headers and bodies in STREAMED or FULL_DUPLEX_STREAMED modes and configuring
+    // the ext_proc filter to retain the content length header. In such cases the side stream
+    // server is responsible to correctly set the content length. If no content length is
+    // provided as part of a header mutation ext_proc falls back to the default cases for
+    // STREAMED or FULL_DUPLEX_STREAMED modes.
     // In http1 codec, removing content length will enable chunked encoding whenever feasible.
     return (
-        body_mode_ == envoy::extensions::filters::http::ext_proc::v3::ProcessingMode::STREAMED ||
-        body_mode_ ==
-            envoy::extensions::filters::http::ext_proc::v3::ProcessingMode::FULL_DUPLEX_STREAMED ||
+        (body_mode_ == envoy::extensions::filters::http::ext_proc::v3::ProcessingMode::STREAMED &&
```

Comment:
This logic can be simplified, like:

if (retain_content_length_header) {
   return false;
}
...

Reply:
Would this also work for the BUFFERED BodySendMode + SKIP Headers? So we could rephrase the whole comment section above?